### ISSUE TITLE
SQL file migrations do not exit on failures

### ISSUE THREAD

**Issue by awinder** [kunklejr/node-db-migrate#231]
I'd like to get some information on why this is the behavior, or if it's just a bug / oversight.  When creating a sql file migration, you end up with code blocks that look like this:

```
  fs.readFile(filePath, {encoding: 'utf-8'}, function(err,data){
    if (err) return console.log(err);
      console.log('received data: ' + data);

    db.runSql(data, function(err) {
      if (err) return console.log(err);
      callback();
    });
  });
```

So in the case of errors you do get a notification, but the process never exits because the callback is skipped over.  The easy fix for our team has been to replace `return console.log(err);` references with `return callback(err);`, but as a default this doesnt make a lot of sense to me, as it leads to things like hung builds in CI environments.  Any chance of changing to `return callback(err);` for the stubbed-out sql file migrations?

**Comment by wzrdtales** [comment]
There is absolutely a chance to get things changed, especially if they're bugs or undefined behavior.

You can also open a pull request, which I would appreciate, fixing this behavior. The feature itself was also introduced by a pull request https://github.com/kunklejr/node-db-migrate/blame/master/lib/migration.js#L117 and this thing was definitely overseen.
What I want to say, changes are always possible and the easiest way for you to get behavior fixed is to open a pull request and probably the fastest ;)

As I have the chance to ask one thing aside:
To get an Idea which Situation you and your team trys to solve.
Why do you use SQL File migrations instead of the build in migrations? Is there something missing you need and you need to cover this with sql. Using SQL Files results in loosing genericness of migrations, which results in loss of flexibility.
To enhance db-mgirate I would like to know, what exactly you're missing and would need to not use SQL Files.

Thanks in advance for your feedback.

**Comment by wzrdtales** [comment]
To add an information here, this bug is currently priorized low and I'm going to fix this earliest in the v0.11.0 version.
But you're free to open a pull request to make it available in v0.10.0 already.

**Comment by awinder** [comment]
Thanks wzrdtales, I'll get you a PR today for this, just figured I'd check to make sure beforehand :smile: 

We're using sql file migrations right now because it lined up with a previous (homegrown) system that was in place, so that feature really helped us transition into node-db-migrate quickly. I think it was also seen as less-onerous for the dev team since it lined up with the old practice of just a bunch of SQL in files.  We also had this practice of doing data migrations that leveraged SQL functions -- I think this is a great situation to just use node.js logic within the migrations now that we're using node-db-migrate, but SQL files seemed like the quickest / most supported way to transition that stuff in.

Definitely want to move into the built in db migrations shortly, theres some really great features and it should help standardize how we're laying out our database changes (in addition to being a good abstraction layer over straight sql).  Thanks for the great project!

**Comment by wzrdtales** [comment]
If you're using mysql currently you might like looking into my other project
https://github.com/wzrdtales/node-ultimate-migrate

This is a reverse migrator, which has capabilities to reverse a database to migrations. Also it allows to make a diff against the actual migrations and generate from that state the new migrations.

You can look into this, may be it helps you. But please mind, this is still beta. Foreign Keys are actually only partially implemented, they get generated if they're new but there is no template yet for foreign keys which aren't new but have changed in some way.

So if you're using mysql, you could try it, if you haven't used special data types like enum which are currently not supported it should be able to generate working migrations.

**Comment by awinder** [comment]
Wow, that's actually really cool.  I'm tempted to give it a shot just to see what happens :grinning: 

I posted up a PR at #232, thanks for all your help!

**Comment by wzrdtales** [comment]
Yeah just look into it and if something went south open an issue :)

**Comment by wzrdtales** [comment]
bugfix patch published under v0.9.3
